linkcheck/models.py

```diff
@@ -338,14 +338,22 @@ def _check_external(self, tested_url, external_recheck_interval):
                 except:
                     # The external web page is mal-formatted #or maybe other parse errors like encoding
                     # I reckon a broken anchor on an otherwise good URL should count as a pass
-                    self.message = "Page OK but anchor can't be checked"
+                    self.message = f"{self.message}, but anchor can't be checked"
                     self.status = True
 
             if response.status_code in REDIRECT_STATI:
                 # This means it could not follow the redirection
                 self.status = False
             elif response.status_code < 300 and response.history:
-                self.message = ' '.join([str(response.history[0].status_code), response.history[0].reason])
+                initial_message = f"{response.history[0].status_code} {response.history[0].reason}"
+                # Remove the message of the redirection target
+                if ", but " in self.message:
+                    _, anchor_message = re.split(", but ", self.message)
+                    self.message = f"{initial_message}, but {anchor_message}"
+                elif "anchor" in self.message:
+                    self.message = f"{initial_message} and {self.message.lower()}"
+                else:
+                    self.message = initial_message
```

Comment:
Do you think this logic makes sense? I'm not sure what the best way is to combine the information about aorking/brokwn redirects and working/broken hash anchors into one message...

Reply:
Nevermind, I changed the order, so at first the message regarding the status code is calculated, and then the optional hash anchor message is appended (if present), so no need for regex parsing afterwards...